cpp/ql/src/semmle/code/cpp/Declaration.qll

```diff
@@ -416,8 +441,19 @@ abstract class AccessHolder extends Declaration {
 /**
  * A declaration that very likely has more C++ access rights than its
  * enclosing element. This comprises `Class` (they have access to their own
- * private members) along with any target of a `friend` declaration.
- *
+ * private members) along with any target of a `friend` declaration.  For
+ * example `MyClass` and `friendFunction` in the following code:
+ * ```
+ * class MyClass
+ * {
```

Comment:
Here I would add `int privateClassMember;` or some such.  For clarity of presentation, I would and perhaps also wrap both `MyClass` and `friendFunction` with an `EnclosingClass` and then point out that `EnclosingClass` does not have access to `privateClassMember` while both `MyClass` and `friendFunction` do.

Reply:
I see your point, but I wonder if such an example will end up over-complicated.  The goal, after all, is to show which elements are a `DirectAccessHolder`, not to fully explain the features of the C++ language.

Reply:
This is a private class, so don't worry too much about its QLDoc.